src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,105 @@
-import React from 'react';
 import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../react-app-env';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  commentsList, setCommentsList,
+}) => {
+  const [yourname, setYourname] = useState('');
```

Comment:
You can use a single state for this `formState` and make a function to mutate them individually, so when you need to reset it you'll avoid 

Types:
```ts
const formInitialState = {
  name: '',
  email: '',
  comment: '',
};

type FormState = typeof formInitialState;
```

State creation
```ts
  const [formState, setFormState] = useState<FormState>(formInitialState);
  
    const handleInput = (key: keyof FormState, value: string) => {
    setFormState(prev => ({
      ...prev,
      [key]: value,
    }));
  };
```

Input interation:
```ts
        <input
          type="text"
          name="name"
          placeholder="Your name"
          value={formState.name}
          className="NewCommentForm__input"
          onChange={(event) => {
            handleInput('name', event.target.value);
          }}
        />

```